### DISCUSSION TITLE
socket connetion error emit 및 access token 관련 논의

### DISCUSSION THREAD

**Discussion by khk811** [42MVP/back-end#54] [Ideas]
어제 42 Blank팀에 계셨던 ycha님이 chaejkim에게 소켓 인증 관련 질문을 받다가, front-end 코드를 좀 봐주셨는데 back-end 코드 부분에서 아이디어를 주셨습니다. (프론트 코드는 chaejkim이 hyki 님께 따로 얘기를 할 것 같아요)

1. handleConnetion() error emit 하기 
: connection.gateway.ts의 handleConnection()에서 try-catch를 해주셨는데 그 부분의 error를 emit을 해서 front단에서 어떤 에러인지를 인지하면 매우 좋을것 같다는 의견이 있었습니다. 아래 코드는 42 blank 팀의 error handling 부분입니다.
```
  async handleConnection(client: Socket): Promise<boolean> {
    try {
      await this.socketJwtAuthService.verify(client);
    } catch (exception: unknown) {
      this.logger.log(`client connection failed ${exception}`);
      if (exception instanceof Error) {
        client.emit('exception', {
          name: exception.name,
          message: exception.message,
        });
      }
      client.disconnect();
      return false;
    }

    this.logger.log(`client connected ${client.id}`);
    return true;
  }

```
2. access token을 header cookie에서 가져오기
: 현재 front에서 쿠키 안에 있는 토큰을 get해서 request header 내 Authorization 키값에 수동으로 쿠키값을 넣어주고 있습니다. 그런데 쿠키는 브라우저의 책임이고, access token을 쿠키에 저장하고 있으니까, frontend에서 쿠키를 넣어줄 게 아니라 request header에 쿠키가 가도록 설정을 하는건 어떤지에 대한 얘기가 나왔습니다. 그렇게되면 프론트에서 withCredential 설정을 해주면, 백에서 request header 내 쿠키 value값을 parsing을 해서 사용하게 됩니다.

**Comment by sinryuji** [comment]
넵 1번은 이미 진행하려던 사항입니다 커넥션 뿐만 아니라 다른 이벤트들이 관련해서도 에러 이벤트를 정의해야 할 거 같아요
2번은 Authorization에 담는게 국룰이라 그렇게 한거긴 한데... 말씀하신 방법대로 해도 상관없아요
